CDP4WebServices.API/Services/ExchangeFileProcessor.cs

```diff
@@ -458,5 +469,103 @@ private Stream ReadStreamFromArchive(ZipEntry zipEntry, string archivePassword)
+        }
+
+        /// <summary>
+        /// Read migration.json file from stream
+        /// </summary>
+        /// <param name="memoryStream">The zip archive stream <see cref="MemoryStream" /></param>
+        /// <param name="password">The password</param>
+        /// <returns></returns>
+        private IList<MigrationPasswordCredentials> ReadMigrationJsonFromStream(MemoryStream memoryStream, string password)
+        {
+            var credentials = new List<MigrationPasswordCredentials>();
+
+            try
+            {
+                using (var zip = ZipFile.Read(memoryStream))
+                {
+                    var migrationJsonZipEntry =
+                        zip.Entries.SingleOrDefault(x => x.FileName.EndsWith("migration.json"));
+
+                    if (migrationJsonZipEntry != null)
+                    {
+                        using (var stream = this.ReadStreamFromArchive(migrationJsonZipEntry, password))
+                        {
+                            credentials = CreateCredentialsList(stream);
+                        }
+                    }
+                }
+            }
+            catch (Exception ex)
+            {
+                var msg = $"Failed to load file. Error: {ex.Message}";
+                Logger.Error(msg);
+            }
+
+            return credentials;
+        }
+
+        /// <summary>
+        ///
+        /// </summary>
+        /// <param name="stream"></param>
+        /// <returns></returns>
+        private static List<MigrationPasswordCredentials> CreateCredentialsList(Stream stream)
+        {
+            var credentialsList = new List<MigrationPasswordCredentials>();
+
+            using (var reader = new StreamReader(stream))
+            {
+                try
+                {
+                    var content = reader.ReadToEnd();
+                    var parsedContent = JObject.Parse(content);
+
+                    if (parsedContent?["credentials"] != null)
+                    {
+                        foreach (var children in parsedContent["credentials"].Children())
+                        {
+                            if (!(children is JProperty property))
+                            {
+                                continue;
+                            }
+
+                            var password = (property.First["password"] as JValue)?.Value.ToString();
+                            var salt = (property.First["salt"] as JValue)?.Value.ToString();
+                            var serverSalt = parsedContent?["server"] != null ? (parsedContent?["server"]["salt"] as JValue)?.Value.ToString() : string.Empty;
+                            credentialsList.Add(new MigrationPasswordCredentials(new Guid(property.Name), password, salt, serverSalt));
+                        }
+                    }
+                }
+                catch (Exception ex)
+                {
+                    Logger.Error($"Failed to load file. Error: {ex.Message}");
```

Comment:
here we are logging an error, but the process continues.. what can go wrong that may cause to have invalid data on the server? If this file fails, i guess the server data is not usable?

Reply:
Correct(error was catched and hidden), but basically inside this flow in the beginning all the passwords have been reset to default: <code>person.Password = AppConfig.Current.Defaults.PersonPassword</code>, but the user who expects migration with passwords it's not notified.